src/AffectList.cpp

```diff
@@ -5,32 +5,21 @@
 #include <range/v3/iterator/operations.hpp>
 
 AFFECT_DATA &AffectList::add(const AFFECT_DATA &aff) {
-    auto new_aff = new AFFECT_DATA{aff};
-    new_aff->next = first_;
-    first_ = new_aff;
-    return *new_aff;
+    data_.insert(data_.begin(), aff);
+    return data_.front();
 }
 
 void AffectList::remove(const AFFECT_DATA &aff) {
-    if (&aff == first_) {
-        first_ = aff.next;
-    } else {
-        AFFECT_DATA *prev;
-
-        for (prev = first_; prev != nullptr; prev = prev->next) {
-            if (prev->next == &aff) {
-                prev->next = aff.next;
-                break;
-            }
-        }
-
-        if (prev == nullptr) {
-            bug("Affect_remove: cannot find paf.");
-            // TODO may leak
-            return;
-        }
+    auto index = &aff - data_.data();
```

Comment:
that's interesting, but I guess it works because vector always has things in a consistent order? Or is std::distance a "safer" way to do it?

Reply:
`std::distance` works on iterators, not references. I'm sure there's a better way to do this: this totally relies on the vector being contiguous.